tools/antithesis-go-instrumentor/common/files.go

```diff
@@ -195,3 +204,26 @@ func ValidateDirectories(input, output string) (err error) {
 	}
 	return
 }
+
+// PathFromBaseDirectory gets the path of someDir relative to baseDir
+//
+// Example:
+// PathFromBaseDirectory("/home/ricky/etcd", "/home/ricky/etcd/server/test")
+//
+//	==> "server/test"
+func PathFromBaseDirectory(baseDir, someDir string) string {
+	baseNorm := CanonicalizeDirectory(baseDir)
+	someNorm := CanonicalizeDirectory(someDir)
+	if baseNorm == someNorm {
+		return ""
+	}
+	someOffset := someNorm
+	if strings.HasPrefix(someNorm, baseNorm) {
```

Comment:
this can be a can of worms.  imagine the following scenario:

/path/to/dir
/path/to/dir_xyz

the second one will look like the prefix of the first one even though it is not a subdirectory. This is a non-trivial thing to test for (see https://github.com/golang/go/issues/18358). Probably the simplest way that works 99% of the time is:

```go
filepath.match(filepath.join(baseNorm,"*"), someNorm)
```

Reply:
Done - thank you.